qemu/tests/win_virtio_driver_update_by_installer.py

```diff
@@ -0,0 +1,109 @@
+                                         cdrom_virtio)
+        logging.info("Changing virtio iso image to '%s'", virtio_iso)
+        vm.change_media("drive_virtio", virtio_iso)
+
+    devcon_path = params["devcon_path"]
+    installer_pkg_check_cmd = params["installer_pkg_check_cmd"]
+    run_install_cmd = params["run_install_cmd"]
+    media_type = params["virtio_win_media_type"]
+
+    # gagent version check test config
+    qemu_ga_pkg = params["qemu_ga_pkg"]
+    gagent_pkg_info_cmd = params["gagent_pkg_info_cmd"]
+    gagent_install_cmd = params["gagent_install_cmd"]
+    gagent_uninstall_cmd = params["gagent_uninstall_cmd"]
+
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+
+    expected_gagent_version = install_gagent(session, test,
+                                             qemu_ga_pkg,
+                                             gagent_install_cmd,
+                                             gagent_pkg_info_cmd)
+
+    uninstall_gagent(session, test, gagent_uninstall_cmd)
+
+    if params.get("check_qemufwcfg", "no") == "yes":
+        driver_name_list.append('qemufwcfg')
+
+    win_uninstall_all_drivers(session, test, params)
+    change_virtio_media(params["cdrom_virtio_downgrade"])
+
+    session = vm.reboot(session)
+
+    if params.get("update_from_previous_installer", "no") == "yes":
+        error_context.context("install drivers from previous installer",
+                              logging.info)
+        install_test_with_screen_on_desktop(vm, session, test,
+                                            run_install_cmd,
+                                            installer_pkg_check_cmd,
+                                            copy_files_params=params)
+    else:
+        for driver_name, device_name, device_hwid in zip(driver_name_list,
+                                                         device_name_list,
+                                                         device_hwid_list):
+            win_driver_utils.install_driver_by_virtio_media(session, test,
+                                                            devcon_path,
+                                                            media_type,
+                                                            driver_name,
+                                                            device_hwid)
+        install_gagent(session, test, qemu_ga_pkg, gagent_install_cmd,
+                       gagent_pkg_info_cmd)
+
+    error_context.context("Upgrade virtio driver to original",
+                          logging.info)
+    change_virtio_media(params["cdrom_virtio"])
+    install_test_with_screen_on_desktop(vm, session, test,
+                                        run_install_cmd,
+                                        installer_pkg_check_cmd,
+                                        copy_files_params=params)
+    session = vm.reboot(session)  # workaround as Bug 1847856
```

Comment:
If need modification in the future when the bz is fixed, better add `# TODO` to mark it.

Reply:
Many thanks @vivianQizhu for all your suggestions ，all are good for me. will update it, and run each guests .